types/encryption.go

```diff
@@ -49,39 +52,32 @@ func Encrypt(pk *rsa.PublicKey, plainText []byte) ([]byte, error) {
 // PemToPrivateKey return rsa private key from pem
 func PemToPrivateKey(skPem []byte) (*rsa.PrivateKey, error) {
 	block, _ := pem.Decode(skPem)
-	// nolint
-	enc := x509.IsEncryptedPEMBlock(block)
-	b := block.Bytes
-	if enc {
-		var err error
-		// nolint
-		b, err = x509.DecryptPEMBlock(block, nil)
-		if err != nil {
-			return nil, errors.Wrap(err, "Failed to decrypt private key")
-		}
+	if block == nil {
+		return nil, errors.New("failed to decode PEM")
 	}
-	parsedSk, err := x509.ParsePKCS1PrivateKey(b)
+
+	// Parse key as PKCS8
+	parsedSk, err := x509.ParsePKCS8PrivateKey(block.Bytes)
```

Comment:
AFAIK PKCS8 isn't backward compatible with PKCS1.. wouldn't this break stuff?

Reply:
Correct
It depends on how these functions are being used. From your above comment, it seems that it would break indeed. So, I'll switch back to PKCS1